Complete Re-work of Chef-Bareos Cookbook

GitBytes opened the pull request:
@sitle 

I have been working on this cookbook over the past nine months or so. It is now to the point (other than a couple of README bits I still need to fill in) where I feel it is worth having you take a look to see if you would like to get this design of the cookbook up and going on the supermarket. I am happy to finish the remaining README bits completed before moving forward but I wanted to get this to you to look over if you have time.

I am happy to have a discussion on things as needed. This is not a small merge but it is much more than it was in the past I think. Let me know if you get time to look through it. I am currently using this in production, pulling directly from our repo for now. If we can get this on the supermarket in the next couple of weeks great (assuming the README gets completed) but overall I am in no rush. Thanks!

(FYI Example Hashes are included in the attributes and tests work via kitchen. ChefSpec tests need a bit of work yet to complete, I also just want more spec tests in general since they are lacking at the moment but I haven't had enough time to retrofit them yet.)

BR
-Ian

GitBytes committed:
bump again for berks

GitBytes committed:
missed this

GitBytes committed:
typo fix

GitBytes committed:
moving to el71 testing

GitBytes committed:
ordering issue in the role

GitBytes committed:
redesigned roles to better suit searching and wrappers

GitBytes committed:
adding new required role for searching, also adds feature to split out storage

GitBytes committed:
incorporating new changes into kitchen tests

GitBytes committed:
Adding new searchable role to better suit wrappers

GitBytes committed:
refactoring for use with roles again, better for wrappers

GitBytes committed:
bumping for role refactor

GitBytes committed:
removing these old tests

GitBytes committed:
Disabling another rubocop cop

GitBytes committed:
correction to node attribute

GitBytes committed:
bump for sane attribute fix

GitBytes committed:
missed this at some point, should match attribute

GitBytes committed:
fixing this to match attribute

byteybyte committed:
Adding unit tests per lamont

Contributing unit tests put forth by @lamont

GitBytes committed:
rubocop cleanup

GitBytes committed:
unit test still not complete, backing out until i have time to focus on them

GitBytes committed:
removing fedora19 tests until i can sort out postgresql mem conf issues

GitBytes committed:
bump with kitchen fedora19 change

GitBytes committed:
Missing the .0 at end of platform url attribute

GitBytes committed:
adding in virtual full testing for client configs

GitBytes committed:
adding lots of new client config attributes, See README

GitBytes committed:
rework of host based pools and how they land in template

GitBytes committed:
reworking several pieces of this template (pools,message,main fileset,attributes)

GitBytes committed:
making message params better, minor edit

GitBytes committed:
better message resource

GitBytes committed:
reworking host pools, adding vfulls (initial), adding host fileset

GitBytes committed:
foodcritic cleanup

GitBytes committed:
adding vfulls and reworking several aspects of pools

GitBytes committed:
needed to retain the @bareos_client for incoming future clients

GitBytes committed:
bump for bugfix to client template

GitBytes committed:
attempting a fix for remote client fd password mismatch bug

GitBytes committed:
bump for more client based bugfix bits

GitBytes committed:
Did not specify attribute necessary

GitBytes committed:
playing with client attribute authority (node/server)

GitBytes committed:
trying this instead for logic for host-pools

GitBytes committed:
bump for attribute logic change

GitBytes committed:
playing with pool logic again, works in kitchen, issues with live

GitBytes committed:
interesting

GitBytes committed:
reverting default pool names, possible collision with deep merging attributes

GitBytes committed:
aligning labels

GitBytes committed:
bump for volume naming change

GitBytes committed:
needed to change this

GitBytes committed:
Trying to make cookbook more general/dynamic

GitBytes committed:
reducing necessary attributes

GitBytes committed:
rearchitecting cookbook to use hases, still work to be done on this

gitter-badger committed:
Add Gitter badge

GitBytes committed:
Merge pull request #6 from gitter-badger/gitter-badge

Add a Gitter chat badge to README.md

GitBytes committed:
starting on a skeleton hashing framework

GitBytes committed:
rubocop cleanup

GitBytes committed:
misplaced, removing

GitBytes committed:
Pulling in the contrib repo bits from 1.0.0 branch

GitBytes committed:
adding in necessary attributes to handle contrib repo

GitBytes committed:
fixing repos, adding conrib, adding example hashes for dir

GitBytes committed:
cleaning up some left overs that won't be needed, adding a reload dir/restart split resources depending

GitBytes committed:
needs testing but should add based on hashing now I think

GitBytes committed:
Key fail, fixed

GitBytes committed:
still not there I think but idea is there mostly

GitBytes committed:
able to cut down on number of static attributes now

GitBytes committed:
reverting slightly, adding in client conf hashes

GitBytes committed:
reworking this a bit to do simpler hashing like pools

GitBytes committed:
foodcritic cleanup

GitBytes committed:
getting in some basics so the templates match their true purpose

GitBytes committed:
prepping some more naming structures

GitBytes committed:
syncing variables with what is used in templates

GitBytes committed:
adding future work for myself, and doing some variable work again

GitBytes committed:
bumping to 2.0.1 so tagging doesn't conflict with branch naming convention

GitBytes committed:
pushing a working model on cookbook, this uses hashing methods for dynamic configs, more work needed but working now

GitBytes committed:
forgot to do lint before last push

GitBytes committed:
Removing quotes on sd port, was breaking things

GitBytes committed:
Various cleanup bits, comment cleanup, resource notifies/subscribes changes

GitBytes committed:
cleanup, organization updates, consistency issues

GitBytes committed:
fixing typo

GitBytes committed:
fixing some premissions

GitBytes committed:
found bug with jobs/jobdefs per client, also don't want sched on restore

GitBytes committed:
typos

GitBytes committed:
Typo bump

GitBytes committed:
hack until I can get logic right

GitBytes committed:
typo

GitBytes committed:
needed another layer to handle deep hash

GitBytes committed:
bumping to handle deeper hashing

GitBytes committed:
I think this should be the fix

GitBytes committed:
There, now that works, make sure to document the lists requirement in the hashing for filesets

GitBytes committed:
fix for fileset bug

GitBytes committed:
fixing a template bug for subscriptions in dir config

GitBytes committed:
adding some tests for autochanger configs

GitBytes committed:
adding an example autochanger device hash

GitBytes committed:
moving the autochanger stuff to its own recipe for ease of edits

GitBytes committed:
adding in some autochanger config bits, work in progress but testing well so far

GitBytes committed:
adding new major feature, tape/autochanger support, work in progress

GitBytes committed:
Refactoring for ease of use and readability. Plus..it works

GitBytes committed:
adding working base support for autochangers, still semi-experimental but functional, will flush out more issues as they come along

GitBytes committed:
putting temporary limitation on postgresql cookbook

GitBytes committed:
minor change in how default attrs for autochangers work

GitBytes committed:
adding an unmanaged host to test against

GitBytes committed:
adding an unmanaged host to test against

GitBytes committed:
adding ability to add custom jobs and unmanaged hosts to appropriate templates

GitBytes committed:
adding section to template to add unmanaged clients if desired, shouldn't fail on nil

GitBytes committed:
adding section to template to add custom jobs if desired, shouldn't fail on nil

GitBytes committed:
bump for custom jobs and unmanaged hosts feature

GitBytes committed:
typo correction

GitBytes committed:
bumping for typo fixes

GitBytes committed:
doing a slight restructuring of client jobs so more flexible

GitBytes committed:
formatting

GitBytes committed:
formatting, cleanup, restructure of client jobs for flexibility

GitBytes committed:
cleanup, formatting, adding some flexibility to client jobs

GitBytes committed:
adding in new attribute to split the client local name versus network name possible differences

GitBytes committed:
working on a template bug, this may be fix

GitBytes committed:
editing attributes to handle test cases better

GitBytes committed:
updating test logic wrapping so I can handle nil entries better

GitBytes committed:
adding nil wrapper logic to template sections for jobs

GitBytes committed:
updating nil wrapper logic and test attribute logic

GitBytes committed:
pushing a fix to my template

GitBytes committed:
bumping for job template and attribute fixes, and client template typo

GitBytes committed:
just they way it worked out, needed to form things in this manner

GitBytes committed:
found typo, missing quotes

GitBytes committed:
had to add an additional layer to format things correctly

GitBytes committed:
removing attribute, didn't need it afterall, adding newline to jobs template

GitBytes committed:
adding a space between config sections

GitBytes committed:
missed more quotes

GitBytes committed:
bump for cleanup

GitBytes committed:
trying something closer to original method

GitBytes committed:
trying to simplify the jobs feature

GitBytes committed:
trying to simplify the jobs feature

GitBytes committed:
Adding back the alternate client name for differnet network name cases

GitBytes committed:
This was messing with wrappers

GitBytes committed:
minor adjustment to test cases

GitBytes committed:
minor adjustment to test cases

GitBytes committed:
adding another net_name attribute for custom setups

GitBytes committed:
migrating dir_helper.conf to template so i can add inclusions for wrappers if desired, like other conf files

GitBytes committed:
bump for dir_helper template conversion

GitBytes committed:
i think this may be a minor bug, testing

GitBytes committed:
missed this from last addition, missing quote

GitBytes committed:
adding default def special for the catalog, can change in wrapper if desired

GitBytes committed:
fixing some permissions that seem to cause issues

GitBytes committed:
Removing the postgresql cookbook constraint of '< 4.0.0'

GitBytes committed:
adding the postgresql cookbook limitation until some things are cleaned up post 4.0 version

GitBytes committed:
adding a sort to the results of the bareos_clients search, fixing attribute naming schema for helper

GitBytes committed:
adding same functionality for the sd as I did with the server/director

GitBytes committed:
adding basic example attribute for the sd_helpoer bits

GitBytes committed:
adding better headers to my templates and had an extra # in there for some reason

GitBytes committed:
bump for adding sd_helper template bits and better headers and schema improvement for dir_helper

GitBytes committed:
attempting to fix a sorting issue that leads to a restart of the director, killing jobs

byteybyte committed:
decided to sort on the template side, seems easier that way, kitchen good, more live tests needed

byteybyte committed:
forgot to update the unmanaged_clients link to server recipe

GitBytes committed:
adding a reload action as an optional feature, still restart by default

GitBytes committed:
This cop has been updated, found from TraveCI Build failure

GitBytes committed:
bumping centos7.1 to 7.2 so less updating is needed for testing, and it is current

GitBytes committed:
this was making wrapper test a hassle

GitBytes committed:
bumping to version 15.2 of bareos by default, fixing my repos for fedora, cleanup for Travis

GitBytes committed:
missed a spot for capitalize

GitBytes committed:
removing a default unmanged host, messing with wrapper tests

GitBytes committed:
Bumping because 15.2 in centos6.7 needs epel repos for tests

GitBytes committed:
missed a few runlist lines in kitchen file

GitBytes committed:
typo fix

GitBytes committed:
typo fix bump

GitBytes committed:
fixing kitchen hashing examples, verify dir is started before reloading config, ver bump again

GitBytes committed:
director attribute not server

GitBytes committed:
attempting some rubocop cleanup for TravisCI

GitBytes committed:
Prepping README for more information later, cleanup basically

GitBytes committed:
Adding some notes to CHANGELOG to give a brief description of current state of cookbook and precautions when upgrading to newest version

GitBytes committed:
formatting

GitBytes committed:
more formatting

GitBytes committed:
Fixing a couple minor typos

GitBytes committed:
adding the gitter badge from the master branch

GitBytes committed:
attempting a side-by-side

GitBytes committed:
adding a description for each set of attributes and adding a better to prep for more info

GitBytes committed:
minor changes to wording

GitBytes committed:
adding placeholder for areas that need filled in with specifics and examples

GitBytes committed:
Got the variable rubocop warnings cleaned up finally

GitBytes committed:
cleaning up some unused attributes, leftovers from old versions

GitBytes committed:
cleaning up my integration testing a bit, adding some alpha unit testing

GitBytes committed:
This is starting to work for me a little better, getting some os seperation as well

GitBytes committed:
need to plan these tests out a bit more, skeleton in place, needs work..

GitBytes committed:
adding lamont's unit test from late last year finally..

GitBytes committed:
removing placeholder

GitBytes committed:
Lamont's test=awesome, my test..not there.. still, success is good

GitBytes committed:
removing unnecessary tests

GitBytes committed:
renaming the primary integration test
Only needed one main integration test in the end

GitBytes committed:
removing unnecessary tests

GitBytes committed:
removing unnecessary tests

GitBytes committed:
already working on some better chefspec tests, these can go now.

GitBytes committed:
merging both tests together to just make an aio server

GitBytes committed:
not reliant on tailor testing that I am aware of, serverspec and chefspec

GitBytes committed:
I am not sure they still want notified on travis testing going forward

GitBytes committed:
not using thor in testing that I am aware of

GitBytes committed:
Useless file, already deleted in another branch as well..

GitBytes committed:
Attribute file cleanup

* adding some test jobs to kitchen integration test
* relabeling default/example section descriptors

GitBytes committed:
Metadata updates

 * Adding some pessimistic cookbook dependency locks
 * Version bump for cleanup and minor bug fixes

GitBytes committed:
Updating the autochanger template for readibility

GitBytes committed:
adding in some additional testing via chefspec

GitBytes committed:
moving the .rspec file to top dir of cookbook for ease of use

GitBytes committed:
updating rubocop config

GitBytes committed:
giving a simpler message to indicate what configs work..

GitBytes committed:
Found some potential dependency issues, making requirements more relaxed but pessimistically.

GitBytes committed:
Higher than 2.0 should be ok, patch bump for now..

GitBytes committed:
Updating test suite to newest chef-client

GitBytes committed:
Removing tests for outdated 12.04 Ubuntu, Removing Fedora Tests

GitBytes committed:
There is already another config line that specifies LineLength cop, redundant

GitBytes committed:
Updating the travis tests to test against latest stable ruby versions

GitBytes committed:
Removing foodcritic test limitations, should be testing against all of it at this point.

GitBytes committed:
Adding back the original maintainer email addresses, was temporary

GitBytes committed:
Cleanup

GitBytes committed:
I don't think we need this in here anymore..

GitBytes committed:
Trying to address the missing line I am seeing, may not be any actual changes here..

GitBytes committed:
Adding a new graphite feature and moving to version 4.0+ for postgresql cookbook

GitBytes committed:
General Cleanup

GitBytes committed:
Removing support for Fedora

GitBytes committed:
Moving up dependency locks

GitBytes committed:
Cleanup clutter

GitBytes committed:
Seemed better to generalize this for use in other places than just the yum repo..

GitBytes committed:
Adding new bits for the graphite plugin recipe

GitBytes committed:
Header cleanup and name correction per LICENSE file

GitBytes committed:
Header cleanup, name correction per LICENSE file

GitBytes committed:
Adding sensitive due to passwords being put into files

GitBytes committed:
Starting the service along with enabling it seems like a basic thing here..

GitBytes committed:
Header cleanup, name correction per LICENSE file.

GitBytes committed:
Adding new graphite plugin recipe:

 * Adds recipe to install plugin related bits
 * Includes a template to auto-populate fields based on attributes

GitBytes committed:
Change to the generic naming attribute for these entries so not yum specific

GitBytes committed:
Bareos repos not compliant with distribution version naming

 * Newest versions of apt cookbook trying to force add a distribution string (i.e. wheezy), this nullifies that issue.

GitBytes committed:
Try to make chef run less noisy, protects from password leaks.

GitBytes committed:
bareos-dir service needs to be started before reload happens otherwise things hang.

GitBytes committed:
Make sure to restart bareos-dir if postgresql makes a change or restarts

GitBytes committed:
Make chef run less noisy, protect from chef log password leaks

GitBytes committed:
Let's start the service as well as enable it.

GitBytes committed:
chef less noisy, protect from chef log pass leak

GitBytes committed:
Adding the graphite plugin to test suite.

GitBytes committed:
Merge remote-tracking branch 'origin/2.0.0'

README Conflict resolved to 2.0.0 branch version

# Conflicts:
#	README.md

GitBytes committed:
Cleanup, adding recipe descriptions, more to add soon hopefully.

GitBytes committed:
Quick fix to include some metadata fits I have been meaning to add

 * source_url
 * issues_url

# Conflicts:
#	.kitchen.yml
#	README.md
#	attributes/default.rb
#	metadata.rb
#	recipes/repo.rb
#	recipes/server.rb
#	recipes/workstation.rb
#	templates/default/bareos-dir.conf.erb
#	templates/default/bareos-fd.conf.erb
#	templates/default/bareos-sd.conf.erb
#	templates/default/bconsole.conf.erb
#	templates/default/client.conf.erb

GitBytes committed:
Quick fix to include some newer metadata bits

 * source_url
 * issues_url

GitBytes committed:
Merge branch 'master' of https://github.com/EMSL-MSC/chef-bareos

GitBytes committed:
Linters didn't like the non-ascii characters, cleaned that up

GitBytes committed:
Fix non-ascii characters for linters

GitBytes committed:
haven't tested yet but linters picked up that these weren't used

GitBytes committed:
fixing chefspec tests for travisci

GitBytes committed:
minor fixes for travis again..

GitBytes committed:
attempting to fix travis ci rspec tests

GitBytes committed:
removing redundant script action

GitBytes commented on CHANGELOG.md:
```diff
@@ -1,11 +1,39 @@
+--------------------
 
-This file is used to list changes made in each version of the bareos cookbook.
+2.2.13
```
I need to add some kind of entry for the current version before this should be merged. I'll give an explanation as to why I did the rework and going forward what the plan may entail...

treenerd commented:
Wow, thats a lot of work! Amazing. Thank you very much.

sitle commented:
It's definitively awesome, thx for your work :p 

I need to test a little bit more for myself, but i'm fine to merge it and update the supermarket. What do you think ?

Best regard.

GitBytes commented:
@sitle @treenerd 

Wow! I am really pleased you guys like it so far! I am itchy to pull the trigger to update the supermarket but I think I would still like to hold off until I get the docs more complete-ish. Thank you so much for the quick offer though @sitle. As soon as I can find a tad more time this week I will try and polish the docs up for a supermarket release, including the README and the CHANGELOG (summarized). I am 50/50 whether to merge this request anyways to the develop branch then still finish docs before going to market (master). I would say if you guys would like to merge #15 but not update the market (master branch) until I can get a docs merge request in the works I think that would work. @treenerd Feel free to chip in there if you would like. Once you both have had enough time to test things and see if it is ok as is, feel free to merge to develop for now. Great to hear from you guys so quickly, cheers!!

-Ian

GitBytes committed:
Updating with a summarized update due to the number of changes

GitBytes committed:
Consistency

GitBytes committed:
Minor update to version for README updates

GitBytes committed:
Updating README

GitBytes committed:
Found a deprecated attribute, file a ticket if this was not right

GitBytes committed:
Updating year for license

GitBytes committed:
lacked some formatting

GitBytes commented:
@sitle @treenerd 
Ok guys.. It should be ready given your tests work out ok. Feel free to merge to both Develop and master and push to the supermarket when ready and Travis-CI tests pass. Cheers!!

-Ian

GitBytes committed:
Yeah...that was needed.. whoops!

Adding back the default['bareos']['storage']['sd_mon_enable'] = 'yes' line I removed in my last few commits

sitle commented:
@GitBytes All done. Thx again for your work.

Best regard.

GitBytes commented:
@sitle Happy to help! I have a tiny bit of cleanup for the README again but I'll wait to push that out until I have something else to get out there.. It is a Supermarket README cosmetic update so low priority.. Formatting got a little wonky I noticed after being posted.. Oh well. Thanks again for getting it up on the Supermarket nonetheless! :) Have a great weekend, cheers!

-Ian
